client/asset/interface.go

```diff
@@ -581,6 +582,15 @@ type Broadcaster interface {
 	SendTransaction(rawTx []byte) ([]byte, error)
 }
 
+// BondDetails is the return from Bonder.FindBond.
+type BondDetails struct {
+	*Bond
+	LockTime time.Time
+	AcctID   account.AccountID
+	BondPKH  []byte
+	PKHFn    func(priv *secp256k1.PrivateKey) []byte
```

Comment:
You could just return a bool from this function if the private key is the one that can refund this bond, and then you don't need to send `BondPKH`.

Reply:
Oh yeah. Would that be better? It could go ahead and make the refund script and log then.

Reply:
idk you would need to give the asset the bondkey then to iterate, or a function that can do that, so I rather like it this way letting core handle the key.

Reply:
You would still iterate over here. Instead of just hashing the private key, the updated `PKHFn` would also check if the result of the has is equal to the PKH. I just thought it was better because the only reason the `BondPKH` comes out of `FindBond` is to go back in, it's never used outside otherwise. It's nbd though.

Reply:
Ohhhh